### ISSUE TITLE
`read_dicom_auto` can return any one of `(Scan | PET | StructureSet | Dose | Segmentation | None)`

### ISSUE THREAD

**Issue by strixy16** [bhklab/readii#32]
`read_dicom_auto` can return any one of `(Scan | PET | StructureSet | Dose | Segmentation | None)`

for `baseImage` I'm assuming it should must return a `Scan` type? 
reason: 

the `baseImage.image` will only work if `baseImage` is of type `Scan`, and we can provide better logging/error handling
```
segMasks = makeMask(
    segImage,
    baseImage.image,
    existing_roi_indices={"background": 0},
    ignore_missing_regex=False,
)
```

if so, should verify this type before proceeding with something like:

```
from imgtools.modules import scan
...
baseImage = read_dicom_auto(baseImageDirPath)
assert isinstance(baseImage, scan.Scan), f"baseImage is not a Scan object. has type {type(baseImage)}"
```

or something

_Originally posted by @jjjermiah in https://github.com/bhklab/readii/pull/31#discussion_r1722185902_

**Comment by strixy16** [comment]
Should update to ```read_dicom_scan``` as this step is meant to load CT or MR and that's the specific function for that.

**Comment by jjjermiah** [comment]
closed by #42
